mgcpy/benchmarks/power_test.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+import numpy as np
+from scipy.spatial.distance import pdist, squareform
+from mgcpy.independence_tests.dcorr import DCorr
+from mgcpy.benchmarks.simulations import w_sim
+from mgcpy.benchmarks.power import power
+
+
+def compute_distance_matrix(data_matrix_X, data_matrix_Y):
+    # obtain the pairwise distance matrix for X and Y
+    dist_mtx_X = squareform(pdist(data_matrix_X, metric='euclidean'))
```

Comment:
Use https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance_matrix.html, instead.

Reply:
this doesn't seem to give us the flexibility to change which metric we use. I think we want to be able to change the metric in the future experiments

Reply:
the function you link allows using different p-norm, but pdist has a larger selection of metric choices, which i think might be good to have

Reply:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.pdist.html#scipy.spatial.distance.pdist